src/Npgsql/NpgsqlConnection.cs

```diff
@@ -736,18 +736,17 @@ internal Task Close(bool async, CancellationToken cancellationToken = default)
                 return Task.CompletedTask;
             }
 
-            return CloseAsync(cancellationToken);
+            return CloseAsync(async, cancellationToken);            
+        }
 
-            async Task CloseAsync(CancellationToken cancellationToken)
+        private async Task CloseAsync(bool async, CancellationToken cancellationToken)
```

Comment:
nit: our coding style is a bit different from the standard one, so no default access modifiers. At some moment I hope it will be aligned with the standard style used in .NET (: